src/lib/homestore_backend/hs_pg_manager.cpp

```diff
@@ -93,6 +94,37 @@ PGManager::NullAsyncResult HSHomeObject::do_create_pg(cshared< homestore::ReplDe
     });
 }
 
+HSHomeObject::HS_PG* HSHomeObject::local_create_pg(shared< ReplDev > repl_dev, PGInfo pg_info) {
+    auto pg_id = pg_info.id;
+    {
+        auto lg = shared_lock(_pg_lock);
+        if (auto it = _pg_map.find(pg_id); it != _pg_map.end()) {
+            LOGW("PG already exists, pg_id {}", pg_id);
+            return dynamic_cast< HS_PG* >(it->second.get());
+        }
+    }
+
+    // create index table and pg
+    // TODO create index table during create shard.
```

Comment:
we may relax this TODO, is there a need to create index during shard create?

Reply:
Good point, based on our current design it seems no longer necessary.